runtime/lua/vim/filetype/detect.lua

```diff
@@ -1322,6 +1322,28 @@ function M.txt(bufnr)
   end
 end
 
+function M.typ(bufnr)
+  if vim.g.filetype_typ then
+    return vim.g.filetype_typ
+  end
+
+  local lines = table.concat(getlines(bufnr, 1, 200))
+
+  if
+    findany(lines, {
+      '^CASE%s%==%s%=SAME$',
```

Comment:
`\=` in Vim is used to match the previous \s 0 or 1 times. In Lua, this does not exist (there is no "?"). So = and %= are the same here. 

The simplest solution would be to check for all of "=", "\s=", "=\s" and "\s=\s" as the middle part. Maybe there's a less verbose version though.

Reply:
Done !

Reply:
From my testing it seems to work as intended